### ISSUE TITLE
Filter pivottable rows on aggregator value

### ISSUE THREAD

**Issue by ClementFromFrance** [nicolaskruchten/pivottable#1043]
Hi,

I'm using pivotUI.
I made my own aggretator which display empty cells or percentage.
I would like to know if there is a way to filter the pivot table in order to remove the empty cells.
Here is the aggregator :
`var DNassortPc = function () {
    return function () {
        return {
            sumDnAssortPc: 0,
            count: 0,
            push: function (record) {
                if (record.presenceneeded == 1) {
                    this.count++;
                    this.sumDnAssortPc += record.dn_assort;
                }
            },
            value: function () { return (this.sumDnAssortPc / this.count) * 100; },
            format: function (x) {
                if (x != undefined && !isNaN(x)) {
                    return x + " %";
                }
            },
            numInputs: 0
        };
    };
};`

and here is a part of the result. I'd like to be able to remove the empty rows.
![image](https://user-images.githubusercontent.com/44844591/48145953-46865700-e2b4-11e8-8e37-6a8dc568f3f8.png)

I hope it is understandable 
thanks for your help

**Comment by Saad55555** [comment]
Hi, can you please apply this function on mtcars data, so that it becomes more clear. Thanks

**Comment by ClementFromFrance** [comment]
> Yup, this is pretty clear, but unfortunately there's no functionality for that, and I can't think of a workaround other than post-processing the table in the `onRefresh` callback to drop the rows you want to drop.
> 
> See also #1044 which seems really similar :)

Thanks for your answer I will develop something and let you know if I manage to do something not so bad :)

**Comment by Saad55555** [comment]
Thanks for the reply, I'm also having an issue in exporting data from rpivotTable, when exported the headers show wrong repetition in excel. You can have a better look to my query in following link:
https://community.rstudio.com/t/headers-rownames-columnnames-wrongly-printed-in-excel-when-rpivottable-data-is-exported/17780

Do you know how to deal with such issue?
Regards

**Comment by Saad55555** [comment]
> > Yup, this is pretty clear, but unfortunately there's no functionality for that, and I can't think of a workaround other than post-processing the table in the `onRefresh` callback to drop the rows you want to drop.
> > See also #1044 which seems really similar :)
> 
> Thanks for your answer I will develop something and let you know if I manage to do something not so bad :)

Hi mate, can you please have a look at the query in the following link:
https://community.rstudio.com/t/headers-rownames-columnnames-wrongly-printed-in-excel-when-rpivottable-data-is-exported/17780
Thanks

**Comment by nicolaskruchten** [comment]
@saad55555 please don’t post off-topic request for people to address you other concerns. You have already raised your issue in the rPivottable repo and stackoverflow, and this issue is not the place to bring it up again.

**Comment by Saad55555** [comment]
Hi mate,
I understand but I’m unable to get an answer from anyplace till now. It’s been more than a week now. And I believe there’s no harm in asking some knowledgeable person for guidance when most of the people refuses to say help. I hope you understand.
Regards

________________________________

**Comment by nicolaskruchten** [comment]
@Saad55555 I do understand, but I'd still like you to comply with this repo's documented [support guidelines](https://github.com/nicolaskruchten/pivottable/blob/master/SUPPORT.md) and not try to hijack existing issues. If you want to open another issue that's fine. That said, the appropriate repo for your concern is the rPivottable repo, and you already have an issue open there. If no one is responding then either they are busy or there is no clear way to accomplish what you are trying to do.

**Comment by ClementFromFrance** [comment]
Hi Nicolas,

Once again I'm asking for your help
I'm trying to use the onRefresh as you suggested. But it's easy to remove the rows which contain data I want to remove, but the problem is with the rowspan of the parent row.
![image](https://user-images.githubusercontent.com/44844591/48211002-d7266b00-e378-11e8-913c-335714cc5bb2.png)

I think we might be able to fix this by binding a remove event that will calculate rowspan in the renderer or something like it but I can't figure out how and where to start. Do you have any ideas or example ?

Regards,
Clement

**Comment by nicolaskruchten** [comment]
It's a pretty tricky problem for sure. Now that I think about it, perhaps a more straightforward solution would be to provide your own `dataClass` which would be a fork of `PivotData` that just doesn't return those keys to the renderer, that way those rows are never in the output in the first place?

**Comment by Saad55555** [comment]
Thanks Nicolas, I totally agree

________________________________

**Comment by ClementFromFrance** [comment]
Hi,

yeah it could be a solution, but then we lost a chance to make a feature wihch could filter values no ?

**Comment by nicolaskruchten** [comment]
I’m not sure I understand your question, sorry, can you rephrase?

**Comment by Saad55555** [comment]
Hi Nicolas, could you please reply on my new query as well which I posted yesterday, I was expecting a reply from you. Thanks

________________________________

**Comment by nicolaskruchten** [comment]
@Saad55555 I've already asked you not to hijack this thread with your unrelated requests. If you persist, you will find yourself blocked from this repo.

**Comment by ClementFromFrance** [comment]
> I’m not sure I understand your question, sorry, can you rephrase?

Sorry 
What I wanted tosay is by developping a way to filter empty values, we would developp a way to filter all values of the pivottable. The possibility to filter value would be great.

But if we filter the empty rows by creating a custom dataclass to avoid rendering them, we would loose the possibility to filter the values of the pivot.

Is it better explained ? (not a fluent english speaker/writer :) )

**Comment by nicolaskruchten** [comment]
Yes, this is better-explained, thanks! You can write in french btw, I live in Québec ;)

So right now this library supports input-level filtering based on the attribute values, but not output-level filtering based on aggregator-computed-values, and I don't think I can imagine a nice general-purpose UI for that second use-case at the moment, so I'm not that worried about not having a way to do it, personally.
